zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -834,30 +834,31 @@ public int aclCacheSize() {
 
         public List<ProcessTxnResult> multiResult;
 
-        /**
-         * Equality is defined as the clientId and the cxid being the same. This
-         * allows us to use hash tables to track completion of transactions.
-         *
-         * @see java.lang.Object#equals(java.lang.Object)
-         */
         @Override
-        public boolean equals(Object o) {
-            if (o instanceof ProcessTxnResult) {
-                ProcessTxnResult other = (ProcessTxnResult) o;
-                return other.clientId == clientId && other.cxid == cxid;
-            }
+        public boolean equals(Object obj) {
+          if (this == obj) {
+            return true;
+          }
+          if (!(obj instanceof ProcessTxnResult)) {
             return false;
+          }
+          ProcessTxnResult other = (ProcessTxnResult) obj;
+          if (clientId != other.clientId) {
+            return false;
+          }
+          if (cxid != other.cxid) {
+            return false;
+          }
+          return true;
```

Comment:
Aside from the same object short circuit, I think the old check that had `a == o.a && b == o.b` was easier to read than all the extra `if` blocks and `!=` comparisons.

Reply:
OK. Reverted.